src/models/dynamic_system.jl

```diff
@@ -0,0 +1,51 @@
+abstract type DynamicSystemState end
+
+
+get_x(dss::DynamicSystemState) = dss.x
+get_n_states(dss::DynamicSystemState) = length(dss.x)
+function state_eqs(dss::DynamicSystemState,time, mass, inertia, forces, moments, h) end
+
+
+abstract type DynamicSystemStateDot end
+
+
+get_xdot(dssd::DynamicSystemStateDot) = dssd.xdot
+get_dynamic_system_state(dssd::DynamicSystemStateDot) = dssd.dss
+
+
+struct SixDOFAeroEuler{T}<:DynamicSystemState
+    x::SVector{13, T}
+end
+
+SixDOFAeroEuler(x::AbstractVector) = SixDOFAeroEuler(SVector{13}(x))
+
+
+get_n_states(dss::SixDOFAeroEuler) = 13
+get_x_names(dss::SixDOFAeroEuler) = [:tas, :α, :β, :ϕ, :θ, :ψ, :p, :q, :r, :x, :y, :z, :pow]
+
+
+function state_eqs(dss::SixDOFAeroEuler, time, mass, inertia, forces, moments, h, pow_dot)
+
+    xdot = sixdof_aero_earth_euler_fixed_mass(
+         time,
+         get_x(dss),
+         mass,
+         inertia,
+         forces,
+         moments,
+         h
+    )
+
+    xdot = [xdot..., pow_dot]
+
+    return SixDOFAeroEulerDot(dss, SVector{13}(xdot))
+end
+
+
+struct SixDOFAeroEulerDot{S<:DynamicSystemState, T}<:DynamicSystemStateDot
+    dss::S
+    # TODO: SVector could probably be inefered from T
```

Comment:
Try to infer size from `DynamicSystemType`

Reply:
solved with https://github.com/AlexS12/FlightMechanicsSimulator.jl/pull/34/files#diff-aea2ebae4d0b00276052aef19770e0057d3eacdd84c65f28ebdf90b4af5885c4R29